### PR TITLE
Add QUIC Workflow

### PR DESCRIPTION
**Pull request by pi-314159**
Add a GitHub Actions workflow that automates the building, testing, and deployment of Docker images with QUIC support.

### CODE REVIEW

File: .github/workflows/quic.yml
Comment (baentsch) [inline .github/workflows/quic.yml:9]: Don't understand this: Why run this on schedule, too? Isn't it sufficient when pushed to "main"? What changes in the code base happen afterwards warranting a re-run?

**Review by baentsch — CHANGES_REQUESTED** [review]
Thanks very much for this CI -- it brings this demo to what I consider the "perfect" usage state (ready-made dockerimage) -- however, it fails to run as it doesn't have access to the OQS docker hub privileges. This is caused IMO by the code being based in your private fork, @pi-314159 . So, can I ask you move this to open-quantum-safe/oqs-demos instead (branch prefixed "pi-", so we know who did it)? You have all required permissions now and CI should then run OK. Please also check the single question above.

File: .github/workflows/quic.yml
Comment (pi-314159) [reply .github/workflows/quic.yml:9]: It uses oqs-provider to generate certificates for nginx. If the scheduled task fails, it indicates a compatibility issue between oqs-boringssl and oqs-provider.

**Comment by pi-314159** [comment]
@baentsch GHCI passed. Please check https://github.com/open-quantum-safe/oqs-demos/actions/runs/10505026939

Also, docker images are available at 
https://hub.docker.com/r/openquantumsafe/curl-quic

https://hub.docker.com/r/openquantumsafe/nginx-quic

**Comment by pi-314159** [comment]
@baentsch GHCI passed. Please check https://github.com/open-quantum-safe/oqs-demos/actions/runs/10505026939

Also, docker images are available at 
https://hub.docker.com/r/openquantumsafe/curl-quic

https://hub.docker.com/r/openquantumsafe/nginx-quic

We need to update the descriptions for these images on Docker Hub

**Comment by baentsch** [comment]
> We need to update the descriptions for these images on Docker Hub

Agreed. This is the exact reason why I created USAGE.md files for the other demos: Their text is then verbatim copied to docker hub. The README*.md are for people building/wanting to understand build options; USAGE*.md for pure users of ready-made images (wanting to understand config/usage options).

**Comment by baentsch** [comment]
> @baentsch GHCI passed. Please check https://github.com/open-quantum-safe/oqs-demos/actions/runs/10505026939

Ah, it was just wrong env variable names. Good.

File: .github/workflows/quic.yml
Comment (baentsch) [reply .github/workflows/quic.yml:9]: Ah, very correct -- good idea! And these potential failures would then be(come) visible at the new QUIC status badge in the README?

**Comment by pi-314159** [comment]
@baentsch

I'm ready to merge this now if you're okay with it.

Regarding the description: If you have access to Docker Hub, could you update the description with links to the following resources?

[nginx README-QUIC.md](https://github.com/open-quantum-safe/oqs-demos/blob/main/nginx/README-QUIC.md) (includes example configuration, configuration file locations, and executable location)
[curl README-QUIC.md](https://github.com/open-quantum-safe/oqs-demos/blob/main/curl/README-QUIC.md) (includes instructions on how to use curl)
These should be clear and sufficient for now.

File: .github/workflows/quic.yml
Comment (pi-314159) [reply .github/workflows/quic.yml:9]: Yes!

**Comment by baentsch** [comment]
> @baentsch
> 
> I'm ready to merge this now if you're okay with it.
> 
> Regarding the description: If you have access to Docker Hub, could you update the description with links to the following resources?
> 
> [nginx README-QUIC.md](https://github.com/open-quantum-safe/oqs-demos/blob/main/nginx/README-QUIC.md) (includes example configuration, configuration file locations, and executable location) [curl README-QUIC.md](https://github.com/open-quantum-safe/oqs-demos/blob/main/curl/README-QUIC.md) (includes instructions on how to use curl) These should be clear and sufficient for now.

Just added those READMEs at docker hub (surprisingly I still have access there... :), but IMO these files are not sufficiently self-explanatory for the "casual user". But we can do it as an experiment to see whether we get questions (and/or any interest at all :) 

Alternatively, again tagging @ajbozarth to consider trying these out and providing feedback and/or USAGE-*quic.md files as part of the [revitalization](https://github.com/open-quantum-safe/oqs-demos/discussions/288) activity.

**Review by baentsch — APPROVED** [review]

**Comment by baentsch** [comment]
I still don't understand why CI fails (incl. the new job). Ideas, @pi-314159 ?

**Comment by pi-314159** [comment]
> I still don't understand why CI fails (incl. the new job). Ideas, @pi-314159 ?

Ha, this PR is from pi-314159/oqs-demos, so it doesn't have access to the Docker credentials.
The CI run that was successful came from open-quantum-safe/oqs-demos/tree/pi-20240822-ghci, which is a fork of pi-314159/oqs-demos/tree/20240822-ghci-quic.

The problem will be resolved once I merge this.

**Pull request merged**
